Add shopify-dev-search extension

d-beck opened the pull request:
## Description

This extension allows you to quickly search the Shopify developer documentation at https://shopify.dev/docs/. I find it useful for looking up the docs for GraphQL mutations and object types, going through Raycast is a lot faster than switching to my browser, opening up the docs page, doing the search and clicking on the results.

## Screencast

https://screen.studio/share/m6oC2YNY

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

nglx reviewed (commented):
Hey @d-beck 👋 
I've added some code comments. Also consider adding keyboard shortcuts for common actions and please look into the Greptile feedback?
🙇

d-beck commented on extensions/shopify-dev-search/src/utilities.ts:
```diff
@@ -0,0 +1,43 @@
+  // Remove <p> tags but keep their content
+  let markdown = html.replace(/<\/?p>/g, "");
+
+  // Convert <strong> tags to markdown bold **
+  markdown = markdown.replace(/<strong>(.*?)<\/strong>/g, "**$1** ");
```
Given the formats of the data returned by the API these two won't matter

d-beck commented:
Hi @nglx, thanks a lot for taking a look at the PR. I've rewritten a good bunch of it support pagination and also addressed the other issues you mentioned and greptile's. Please let me know if anything else needs to be changed, thanks!

d-beck commented:
Hi @nglx, just checking on this, any other changes we should make? Thanks
